src/openrct2/actions/ParkMarketingAction.cpp

```diff
@@ -70,7 +70,7 @@ GameActions::Result ParkMarketingAction::Execute() const
     campaign.Flags = MarketingCampaignFlags::FIRST_WEEK;
     if (campaign.Type == ADVERTISING_CAMPAIGN_RIDE_FREE || campaign.Type == ADVERTISING_CAMPAIGN_RIDE)
     {
-        campaign.RideId = static_cast<ride_id_t>(_item);
+        campaign.rideId = RideId::FromUnderlying(_item);
```

Comment:
What about changing `_item` to be `RideId`?

Reply:
Because its not always a ride id.

Reply:
Alright, out of scope for this PR then.